devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy

```diff
@@ -1035,6 +1163,366 @@ def zwaveEvent(physicalgraph.zwave.commands.applicationstatusv1.ApplicationRejec
+	if (device.currentValue("numberOfSlotsDailyRepeating")) {
+		
+		def result = []
+		def codeID = cmd.userIdentifier
+		def lockCodes = loadLockCodes()
+		def codeName = getCodeName(lockCodes, codeID)
+
+		if (cmd.durationHour == 255 && cmd.durationMinute == 255 && cmd.startHour == 255 && cmd.startMinute == 255 && cmd.weekDayBitmask == 255) {
+			def scheduleType = getScheduleType(codeID)
+			if (isScheduleScanningInProgress(codeID) || scheduleType == "always" || scheduleType == "unknown") {
+				log.trace "[DTH] ScheduleEntryLockDailyRepeatingReport() -  Schedule found empty while scanning for codeID := $codeID"
+				result << createEvent(name: "scheduleChanged", value: "$codeID $SCHEDULE_TYPE_DAILY_REPEATING unset", isStateChange: false, displayed: false)
+				// For Kwikset lock, we know the type of schedule type through confugurationGet() and hence an empty schedule case will never be encountered
+				// For other locks, the schedule type is not known and hence check for year day and week day schedule now
+				if (device.currentValue("numberOfSlotsYearDay")) {
+					log.trace "[DTH] ScheduleEntryLockDailyRepeatingReport() -  Requesting year day schedule for codeID := $codeID"
+					result << response(requestYearDaySchedule(codeID))
+				} else if (device.currentValue("numberOfSlotsWeekDay")) {
+					log.trace "[DTH] ScheduleEntryLockDailyRepeatingReport() -  Requesting week day schedule for codeID := $codeID"
+					clearWeekdayScanningDataFromState(codeID)
+					result << response(requestAllWeekDaySchedule(codeID))
+				} else {
+					// schedule scanning for this codeID is complete
+					setUserInfo([codeID: "" + codeID, phoneNum: "", accessType: "always", lastUpdated: new Date().getTime()])
+					log.trace "[DTH] ScheduleEntryLockDailyRepeatingReport() -  Schedule scanning complete for codeID := $codeID"
+					checkAndScanNextSchedule(codeID)
+				}
+			} else {
+				log.trace "[DTH] ScheduleEntryLockDailyRepeatingReport() -  Schedule deleted successfully for codeID := $codeID"
+				result << scheduleDeleteEvent(codeID)
+				def deviceName = device.displayName
+				def map = [name: "scheduleChanged", value: "$codeID $SCHEDULE_TYPE_DAILY_REPEATING deleted", descriptionText: "Cleared recurring schedule for \"$codeName\"", isStateChange: true, displayed: true]
+				map.data = [lockName: deviceName, notify: true, notificationText: "Cleared recurring schedule for \"$codeName\" in $deviceName at ${location.name}"]
+				result << createEvent(map)
+			}
+		} else {
+			def savedStartTime = state["setDailyRepeatingStartTime$codeID"]
+			def savedEndTime = state["setDailyRepeatingEndTime$codeID"]
+			def savedWeekDayBitmask = state["setDailyRepeatingBitMask$codeID"]
+			
+			def changeType = getScheduleChangeType(codeID, SCHEDULE_TYPE_DAILY_REPEATING)
+			
+			if (savedStartTime && savedEndTime && savedWeekDayBitmask) {
+				def startTime = getWeekDayHoursAndMinutes(savedStartTime)
```

Comment:
Every time you call `getWeekDayHoursAndMinutes` you're immediately breaking it up into these separate variables. You should either change the signature to just return a list and then do:

`def (startHour, startMin) = getWeekDayHoursAndMinutes(savedStartTime)`

or just reference startTime's individual attributes every time.

Reply:
Unfortunately the destructuring assignment syntax `def (startHour, startMin)` is broken by the sandbox parser, or at least was last time I used it. I agree just using startTime.hour is stylistically better than startHour, though.